shop/src/main/java/ro/msg/learning/shop/service/ProductCategoryService.java

```diff
@@ -0,0 +1,37 @@
+package ro.msg.learning.shop.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.model.ProductCategory;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class ProductCategoryService {
+    private final ProductCategoryRepository productCategoryRepository;
+
+    public void saveProductCategory(ProductCategoryDto productCategoryDto) {
```

Comment:
Services should work directly with the model, not the Dto. The mapping from Dto to Model and the other way around should be done in Controller. 

Reason: You should be able to use the service from multiple controllers (applications) independent of the DTO defined by you. The Dto is used just to provide the necessary data to the UI. 

Please do the corresponding changes for all the services on your application.